examples/kubernetes/cilium.yaml

```diff
@@ -104,6 +100,7 @@ spec:
           - "--kvstore-opt"
           - "etcd.config=/var/lib/etcd-config/etcd.config"
           - "--disable-ipv4=$(DISABLE_IPV4)"
+          - "$(PROMETHEUS_METRICS_CONFIG)"
```

Comment:
If the users doesn't create prometheus config map, docker container will be created from:
```
                "args": [
                    "--debug=$(CILIUM_DEBUG)",
                    "-t",
                    "vxlan",
                    "--kvstore",
                    "etcd",
                    "--kvstore-opt",
                    "etcd.config=/var/lib/etcd-config/etcd.config",
                    "--disable-ipv4=$(DISABLE_IPV4)",
                    "$(PROMETHEUS_METRICS_CONFIG)"
                ],
```
to something like:
```
            "--debug=false",
            "-t",
            "vxlan",
            "--kvstore",
            "etcd",
            "--kvstore-opt",
            "etcd.config=/var/lib/etcd-config/etcd.config",
            "--disable-ipv4=false",
            "$(PROMETHEUS_METRICS_CONFIG)"
```

Can't you check on cilium-agent the environment variable `PROMETHEUS_METRICS_CONFIG`?

Reply:
@raybejjani also opened #2243 which would implement this behaviour. I agree with @aanm that this is the cleaner behaviour.

Reply:
@aanm Wouldn't it make `$(PROMETHEUS_METRICS_CONFIG)` empty? This is ugly but won't break older clients. At least, it doesn't cause errors when I try it. More importantly, I used viper.Get in #2234 but I noticed that in #2238 you removed the only other instance and used `os.GetEnv`. Do we have a policy/preference on this?

@tgraf ok. I interpreted your "merge simple now" comment to mean you didn't want the code change in this PR. Since I also think checking the environment variable is the better choice, and this is going around in circles, I will take this decision out of your hands and decide:
I will replace this PR with the code-change version, and @aanm will have to tell me if he doesn't want any `viper.Get` in there.

Reply:
>Wouldn't it make $(PROMETHEUS_METRICS_CONFIG) empty? 

No, if the env variable is not defined t will show as I wrote `"$(PROMETHEUS_METRICS_CONFIG)"`